waitress/app/tests/test_routes.py

```diff
@@ -208,6 +213,23 @@ def test_can_regularize_guest_names(self):
         regularized = regularize_guest_names(guest_list)
         assert regularized[-1].firstname == "Guest 10"
 
+    def test_user_can_view_admin_login_page(self):
+        response = self.client.get("")
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.templates[0].name, "login.html")
+
+    def test_user_redirects_to_login_page_on_login_failure(self):
+        user_data = dict(username="user", password="password")
+        response = self.client.post("", data=user_data, follow=True)
```

Comment:
what does the follow keyword argument do?

Reply:
A redirect occurs so to follow to the new url, you need to pass in that.